### PR TITLE
allow clear_origin in safe xcm builder

### LINKED ISSUE #3770: XCM builder pattern safe should allow clear_origin before buy execution
The [XCM builder pattern](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/procedural/src/builder_pattern.rs) provides builders for making XCM messages:
- builder
- builder_unpaid
- builder_unsafe

These builders just restrict the instructions you can use at the start, they force some instructions before others.
The first is meant to be used by regular extrinsics and rust clients.
The second is meant to be used by system functions.
The third is meant to be used for tests, teaching, tinkering and sometimes when the others do not allow you to do what you need.
I found a case of this that should be easy to solve.

Right now we can't do the following:
```rust
Xcm::<()>::builder()
  .reserve_asset_deposited(..)
  .clear_origin(..) // <- This is the problem
  .buy_execution(..)
  .deposit_asset(..)
  .build();
```
We need to use the `builder_unsafe` variant for it to work.

It is a perfectly safe idea to allow at least one `clear_origin` between an instruction that loads the holding register and one that pays for fees.
The solution is to implement the `clear_origin` method in the type `XcmBuilder<Call, LoadedHolding>` in `polkadot/xcm/procedural`.

### PR DESCRIPTION
**Pull request by orgr**
Fixes #3770 

Added `clear_origin` as an allowed command after commands that load the holdings register, in the safe xcm builder.

Checklist
- [x] My PR includes a detailed description as outlined in the "Description" section above
- [x] My PR follows the [labeling requirements](https://github.com/paritytech/polkadot-sdk/blob/master/docs/contributor/CONTRIBUTING.md#Process) of this project (at minimum one label for T required)
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)

### CODE REVIEW

**Review by franciscoaguirre — COMMENTED** [review]
Looks good! 😁  Just missing a test

File: prdoc/pr_4777.prdoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+title: XCM builder pattern allows clear_origin before buy_execution.
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      Added clear_origin as an allowed command after commands that load the holdings register, in the safe xcm builder.
+      Previously, although it's logically allowed, an XCM could not be built like this:
+      ```rust
+      let xcm = Xcm::builder()
+          .withdraw_asset((Parent, 100u128).into())
```
Comment (franciscoaguirre) [inline prdoc/pr_4777.prdoc:13]: No need for the `.into()` now

File: prdoc/pr_4777.prdoc [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+          .clear_origin
+          .buy_execution((Parent, 1u128).into())
+          .deposit_asset(All.into(), AccountId32 { id: [0u8; 32], network: None }.into())
+          .build();
+      ```
```
Comment (franciscoaguirre) [inline prdoc/pr_4777.prdoc:18]: I'd add "You had to use the unsafe builder

**Comment by franciscoaguirre** [comment]
Looking good! Lets see if CI passes

**Comment by orgr** [comment]
The failure looks like something not related to the new code

**Review by franciscoaguirre — APPROVED** [review]

**Comment by franciscoaguirre** [comment]
Need another review here, @gupnik what do you think?

File: prdoc/pr_4777.prdoc [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+doc:
+  - audience: Runtime Dev
+    description: |
+      Added clear_origin as an allowed command after commands that load the holdings register, in the safe xcm builder.
+      Previously, although it's logically allowed, an XCM could not be built like this:
+      ```rust
+      let xcm = Xcm::builder()
+          .withdraw_asset((Parent, 100u128))
+          .clear_origin
```
Comment (gupnik) [inline prdoc/pr_4777.prdoc:14]:
```suggestion
          .clear_origin()
```

**Review by gupnik — APPROVED** [review]

**Pull request merged**